src/main/telemetry/smartport.c

```diff
@@ -653,48 +653,73 @@ void handleSmartPortTelemetry(void)
                     uint32_t tmpi = 10000; // start off with at least one digit so the most significant 0 won't be cut off
 
                     // ones column
+                    // ARM states are either mutually exclusive or implied by an  other , 	room  to encode additional  mode states within the same ba
                     if (!isArmingDisabled())
-                        tmpi += 1;
-                    else
-                        tmpi += 2;
+                        tmpi = 1;    	// was tmpi += 1
+                    else		// meaning ready to ARM
+                        tmpi = 2;    	// was tmpi += 2
 
                     if (ARMING_FLAG(ARMED))
-                        tmpi += 4;
+                        tmpi = 3;      	// was  tmpi += 4
 
                     // tens column
+                    //   slot for mutually exclusive modes, allow to encode 10  modes within the available bandwidth
                     if (FLIGHT_MODE(ANGLE_MODE))
-                        tmpi += 10;
+                        tmpi = 10;
```

Comment:
@jaymal66 you use `tmpi = value` everywhere, meaning that you will basically report a *single* nuber always, regardless of mode combination. I.e. `ANGLE` & `HEADFREE_MODE` will be reported as `200` not as `210`.  Did you test your code?

Reply:
I concur, this won't work at all.

Reply:
Guys you seems to ignore the comment characterizing the "tens column" slot 
  //   slot for mutually exclusive modes,
Meaning : Each of these mode when active exclude others in the same slot to be also active
Example when Angle mode is active  Passthru can't be.. hence a single number can be used to identify unequivocally each  these "mutually exclusive modes" the state of the Flight controller
This is all about  enumerating efficiently and  unequivocally all the possible states of the FC
So anything wrong about the enumeration process used  in this "tens column" slot could  only be of having included an non mutually exclusive mode in the lot . meaning a mode which can be active as any of the others in the slot.
If we don't agree on that we have a  problem

Reply:
For one: your mutually exclusive assumption is wrong - see my comment a few lines lower. And two: please test how ANGLE & HEADFREE is reported with your code and tell if that's really your intention.